internal/compiler/program.go

```diff
@@ -330,6 +330,11 @@ func (p *Program) getSemanticDiagnosticsForFile(sourceFile *ast.SourceFile) []*a
 			diags = append(diags, checker.GetDiagnosticsWithoutCheck(sourceFile)...)
 		}
 	}
+	for _, directive := range sourceFile.CommentDirectives {
+		if directive.Kind == ast.CommentDirectiveKindNoCheck {
+			return nil
+		}
+	}
```

Comment:
I think this isn't quite right; what we should be doing here is skipping the checker loop above if this condition is true.

Something like:

```go
noCheck := slices.ContainsFunc(sourceFile.CommentDirectives, (d) bool { return d.Kind == ast.CommentDirectiveKindNoCheck }
if !noCheck {
	for _, checker := range p.checkers {
		// ...
	}
}
```

Reply:
(Or maybe SourceFile should have a NoCheck field).

Reply:
Thanks for quick feedback. Updated impl based on your idea.